src/services/userService.js

```diff
@@ -27,14 +30,81 @@ export function getUser(id) {
   });
 }
 
+/**
+ * Get a user from refreshToken
+ *
+ * @param  {Object}  req
+ * @return {Promise}
+ */
+export function getUserFromRefreshToken(req) {
+  return User.where({ refreshToken: req.headers['refresh-token'] }).fetch();
+}
+
+
+/**
+ * Create new user.
+ *
+ * @param  {Object}  user
+ * @return {Promise}
+ */
+export async function createUser(user) {
+  return new User({
+    name: user.name,
+    username: user.username,
+    password: await generateHash(user.password),
+    email: user.email
+  }).save();
+
+}
+
+/**
+ * Returns new the current token
+ *
+ * @param  {Object}  req
+ * @return {Promise}
+ */
+export async function getUserToken(req) {
+  const user = await getUserFromRefreshToken(req);
+
+  const accessToken = generateJWT(user.attributes, process.env.ACCESS_TOKEN_VALIDITY);
+  const refreshToken = generateJWT(user.attributes, process.env.REFRESH_TOKEN_VALIDITY);
+
+  return new Promise(async resolve => {
+    await User.where({ username: user.attributes.username }).save({ refreshToken }, { patch: true });
+    resolve({ accessToken, refreshToken, message: 'tokenized successful' });
+
+  });
+
+}
+
 /**
  * Create new user.
  *
  * @param  {Object}  user
  * @return {Promise}
  */
-export function createUser(user) {
-  return new User({ name: user.name }).save();
+export function loginUser(user) {
+
+  const accessToken = generateJWT(user, process.env.ACCESS_TOKEN_VALIDITY);
+  const refreshToken = generateJWT(user, process.env.REFRESH_TOKEN_VALIDITY);
+
+  return new Promise(async (resolve, reject) => {
+
+    const USER = await new User({ username: user.username }).fetch();
```

Comment:
Don't use UPPERCASE for these constants. They are neither string literals nor numbers.